core/src/main/java/org/apache/druid/timeline/CompactionState.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.timeline;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.indexer.partitions.PartitionsSpec;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class CompactionState
+{
+  private final PartitionsSpec partitionsSpec;
+  // org.apache.druid.segment.IndexSpec cannot be used here to avoid the dependency cycle
```

Comment:
couldn't understand what is the cycle ? .. do you mean `IndexSpec` can contain a `CompactionState` , so json serde would fail ?

Reply:
The thing is `IndexSpec` is in the `processing` module while this class is in the `core` module. Since `processing` has a dependency on `core`, I cannot add a new dependency of `core` -> `processing` since it will introduce a cycle. I updated this comment more understandable.

Reply:
got it, thanks.

maybe in next round of module merge: merge core into processing if there is no use case of anyone depending on druid-core directly.